proto-mailbox/src/test/kotlin/actor/proto/mailbox/DefaultMailboxTest.kt

```diff
@@ -0,0 +1,102 @@
+package actor.proto.mailbox
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+class DefaultMailboxTest {
+
+    @Test
+    fun `check that the messages are delivered in the same order with async dispatcher`() {
+
+        val mailbox = newUnboundedMailbox()
+        assertEquals(MailboxStatus.IDLE, (mailbox as DefaultMailbox).status())
+        val invoker = TestMailboxInvoker()
+        mailbox.registerHandlers(invoker, Dispatchers.DEFAULT_DISPATCHER)
+        mailbox.postUserMessage("TestUserMessage1")
+        mailbox.postUserMessage("TestUserMessage2")
+        mailbox.postSystemMessage(SomeSystemMessage(17))
+        mailbox.postSystemMessage(SomeSystemMessage(18))
+        mailbox.postUserMessage("TestUserMessage3")
+        mailbox.postSystemMessage(SomeSystemMessage(19))
+        Thread.sleep(500) // let the DefaultDispatcher to call mailbox.run()
+        assertEquals("TestUserMessage1", invoker.userMessages[0])
+        assertEquals("TestUserMessage2", invoker.userMessages[1])
+        assertEquals("TestUserMessage3", invoker.userMessages[2])
+        Thread.sleep(100) // let the DefaultDispatcher to call mailbox.run()
```

Comment:
Do we need the sleep or are there any other ways to await completion?

Reply:
The other tool will make the test bigger and clutter the content with redundant test code